packages/babylon/src/parser/expression.js

```diff
@@ -454,7 +454,7 @@ export default class ExpressionParser extends LValParser {
       );
     } else if (this.match(tt.questionDot)) {
       this.expectPlugin("optionalChaining");
-
+      state.optionalChainMember = true;
```

Comment:
Not sure if this is relevant to this case, but is there any issue with `state` possibly being reused and `optionalChainMember` being true when it shouldn't?

Reply:
`optinalChainMember` is `true` only after encountering `?.` so i don't think there might be any issue. Once we encounter `?.` then we are sure that the rest of chain will be `optionalChain Member` with optional value = `false` ( meaning that the node is in `optionalChain` but it is not optional(root of optional chain) ). When we encounter ')' like in `(a?.b).c` then the chain is no longer optional and so state doesn't have `optionalChainMember` property for `(a?.b)` object. Should i also add `(a?.b).c` in test cases to make it more clear or can we make it a good first issue for first timers looking for contribution?

Reply:
@Qantas94Heavy , i added additional test cases, please take a look. Maybe that clear the confusion or is there still some doubt?